doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -595,90 +584,85 @@ the conversion in hypotheses :n:`{+ @ident}`.
+   The behavior can be tuned using the :cmd:`Arguments` command.
 
-.. tacv:: simpl @pattern
+.. tacn:: cbn {? @reductions } {? @occurrences }
 
-   This applies :tacn:`simpl` only to the subterms matching
-   :n:`@pattern` in the current goal.
+   :tacn:`cbn` tactic was intended to be a more principled, faster and more
+   predictable replacement for :tacn:`simpl`.
 
-.. tacv:: simpl @pattern at {+ @natural}
+   :tacn:`cbn` tactic accepts the same reductions as :tacn:`cbv` and
+   :tacn:`lazy`. The behavior can be tuned using the :cmd:`Arguments` command.
 
-   This applies :tacn:`simpl` only to the :n:`{+ @natural}` occurrences of the subterms
-   matching :n:`@pattern` in the current goal.
+   The main difference between :tacn:`cbn` and :tacn:`simpl` is that
+   :tacn:`cbn` may unfold constants even when they cannot be reused in recursive calls:
+   :g:`succ t` is reduced to :g:`S t`.
 
    .. exn:: Too few occurrences.
       :undocumented:
 
-.. tacv:: simpl @qualid
-          simpl @string
+   .. flag:: Debug RAKAM
 
-   This applies :tacn:`simpl` only to the applicative subterms whose head occurrence
-   is the unfoldable constant :n:`@qualid` (the constant can be referred to by
-   its notation using :n:`@string` if such a notation exists).
+      This flag makes :tacn:`cbn` print various debugging information.
+      ``RAKAM`` is the Refolding Algebraic Krivine Abstract Machine.
 
-.. tacv:: simpl @qualid at {+ @natural}
-          simpl @string at {+ @natural}
+.. tacn:: unfold {+, @reference_occs } {? @occurrences }
 
-   This applies :tacn:`simpl` only to the :n:`{+ @natural}` applicative subterms whose
-   head occurrence is :n:`@qualid` (or :n:`@string`).
+   .. todo PR: how does this work with multiple reference_occs and occurrence #s?
 
-.. flag:: Debug RAKAM
-
-   This flag makes :tacn:`cbn` print various debugging information.
-   ``RAKAM`` is the Refolding Algebraic Krivine Abstract Machine.
-
-.. tacn:: unfold @qualid
-   :name: unfold
-
-   This tactic applies to any goal. The argument qualid must denote a
-   defined transparent constant or local definition (see
-   :ref:`gallina-definitions` and
-   :ref:`vernac-controlling-the-reduction-strategies`). The tactic
-   :tacn:`unfold` applies the :math:`\delta` rule to each occurrence
-   of the constant to which :n:`@qualid` refers in the current goal
-   and then replaces it with its :math:`\beta\iota\zeta`-normal form.
-   Use the general reduction tactics if you want to avoid this final
-   reduction, for instance :n:`cbv delta [@qualid]`.
+   Applies :term:`delta-reduction` to
+   the constants specified by each :n:`@reference_occs`
+   and replaces them with their :math:`\beta\iota\zeta`-normal form.
+   If :n:`@occurrences` is given,
+   the specified occurrences will be replaced in the selected hypotheses and/or goal.
+   If :n:`@occurrences` is not given, every occurrence of the constants in the goal is
```

Comment:
`@reference_occs` may restrict which occurrences in the goal are unfolded, e.g., `unfold foo at 1, bar at 2`.